src/openrct2/ride/gentle/ObservationTower.cpp

```diff
@@ -35,11 +35,11 @@ void vehicle_visual_observation_tower(
     int32_t baseImage_id = (vehicle->restraints_position / 64);
     if (vehicle->restraints_position >= 64)
     {
-        if ((imageDirection / 8) && (imageDirection / 8) != 3)
+        if (auto directionOffset = imageDirection / 8; (directionOffset != 0) && (directionOffset != 3))
```

Comment:
needs checked against the assembly. Also we tend not to use the if with initializer.

Reply:
I wrote two functions to get the difference between the assemblies.

First:
```
#include <cstdint>

auto f(int32_t imageDirection){
    return (imageDirection / 8) && (imageDirection / 8) != 3;
}
```
[https://godbolt.org/z/7Yczqe](https://godbolt.org/z/7Yczqe)

Second:
```
#include <cstdint>

auto f(int32_t imageDirection){
    auto directionOffset = imageDirection / 8;
    return (directionOffset != 0) && (directionOffset != 3);
}
```
[https://godbolt.org/z/hz61Te](https://godbolt.org/z/hz61Te)

The diff between assemblies can be seen here: [https://www.diffchecker.com/OJKsMcBr](https://www.diffchecker.com/OJKsMcBr)

I ran these functions in all range of ```int32_t``` and results were identical.
I've noticed that this same technique was already employed here: [LaunchedFreefall.cpp#L61](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/thrill/LaunchedFreefall.cpp#L61)

Reply:
No I mean checked against the original rct2 assembly

Reply:
Here we have artifacts before and after the change:
 - [before](https://github.com/vjavs/OpenRCT2/actions/runs/288461434)
 - [after](https://github.com/vjavs/OpenRCT2/actions/runs/288494654)

But I have no clue how to compare them. Either against each other or against the original rct2 assembly.
Actually I checksum file `libopenrct2.so` and they diverged.

How could we do that? Also, I would love to know why is it necessary to check against the binary in such a condition.

Reply:
It's necessary because it looks wrong so I'd like to make sure that it is doing what it is meant to do. You can check the assembly with a debugger/dissassembler like olydbg or ida or you can just leave it for me to check

Reply:
> or you can just leave it for me to check

Sure, please.

Reply:
![image](https://user-images.githubusercontent.com/1277401/95109118-0e078600-0734-11eb-8eb5-347da5a86389.png)
Yup all looks fine. Could you make it a bit more readable by moving around the if:

```
        auto directionOffset = imageDirection / 8;
        if ((directionOffset == 0) || (directionOffset == 3))
        {
            baseImage_id = vehicleEntry->base_image_id + 8;
        }
        else
        {
            baseImage_id *= 2;
            baseImage_id += vehicleEntry->base_image_id;
            if (directionOffset == 1)
            {
                 baseImage_id += 28;
            }
            else
            {
                baseImage_id += 22;
            }
        }

```

Reply:
Nice check! :)